File: backend/api-gateway/src/types/Insurance.ts
export interface Insurance {
  id: number;
  userId: number;
  firstname: string;
  lastname: string;
  type: InsuranceType;
  from: string;
  to: string;
  vehicleRegistration: string;
  vehicleType: VehicleType;
  vehicleKm: number;
  speedingTickets: number;
  drunkDrivingTickets: number;
}

export enum InsuranceType {
  CAR = 'car',
  HOME = 'home',
  LIFE = 'life',
}

export enum VehicleType {
  Sport = 'sport',
  Sedan = 'sedan',
  SUV = 'suv',
  Truck = 'truck',
  Van = 'van',
  Motorcycle = 'motorcycle',
  Other = 'other',
}

File: backend/insurance-service/pkg/models/insurance.go
package models

import "time"

type Insurance struct {
	ID                  uint       `gorm:"primary_key" json:"id"`
	Firstname           string     `json:"firstname"`
	Lastname            string     `json:"lastname"`
	Type                string     `json:"type"`
	From                *time.Time `json:"from"`
	To                  *time.Time `json:"to"`
	VehicleRegistration string     `json:"vehicleRegistration"`
	VehicleType         string     `json:"vehicleType"`
	VehicleKm           int        `json:"vehicleKm"`
	SpeedingTickets     int        `json:"speedingTickets"`
	DrunkDrivingTickets int        `json:"drunkDrivingTickets"`
}
